x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/Page.java

```diff
@@ -222,6 +223,13 @@ public void writeTo(StreamOutput out) throws IOException {
      */
     public void releaseBlocks() {
         blocksReleased = true;
-        Releasables.closeExpectNoException(blocks);
+        // blocks can be used as multiple columns
+        var set = new IdentityHashMap<Block, Object>();
```

Comment:
`Collections.newSetFromMap` is the idiomatic way to do this. And all the cool kids want to be idiots, right?

Reply:
:)
I actually tried that first but it blew up because the blocks equals implementation checks the content meaning different blocks end up looking equal meaning they are not closed as they are considered a duplicate. 
Which causes a bunch of leaks - hence why I'm using the Identity match alone.